### PR TITLE
dev/core#382 Ensure that no db errors are generated when trying to up…

### PR DESCRIPTION
**Pull request by seamuslee001**
…date a civicrm group linked to an organization when the group id doesn't match the id of the relevant row in civicrm_group_organization

Overview
----------------------------------------
When updating a group linked to an organization. If the record id in the civicrm_group_organization table doesn't match that of the groupID there can be a db error generated because of the unique key on the civicrm_group_organization table

Before
----------------------------------------
DB error generated

After
----------------------------------------
No DB error

ping @mattwire @eileenmcnaughton @nganivet @monishdeb This is especially an issue in Multisite configurations

### CODE REVIEW

File: CRM/Contact/BAO/Group.php [hunk 1]
```diff
@@ -450,6 +450,10 @@ public static function create(&$params) {
     if (!empty($params['organization_id'])) {
       $groupOrg = $params;
       $groupOrg['group_id'] = $group->id;
+      // dev/core#382 Keeping the id here can cause db errors as it tries to update the wrong record in the Organization table
+      if (isset($groupOrg['id'])) {
```
[the remark this answers is not in the document]
Comment (seamuslee001) [reply CRM/Contact/BAO/Group.php]: @eileenmcnaughton for the moment i think i'll just not worry about fixing the passing by reference but have made an update to cover what your after

**Comment by eileenmcnaughton** [comment]
@seamuslee001 OK - sure squash these & merge-on-pass

**Comment by seamuslee001** [comment]
@eileenmcnaughton done

**Comment by seamuslee001** [comment]
Jenkins re test this please

**Pull request merged**